Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -522,6 +522,67 @@ class InAppMessageViewTest: IntegrationTest {
         // If url is valid, check if `handleDeepLink` method is called
         XCTAssertFalse(deeplinkUtilMock.handleDeepLinkCalled)
     }
+
+    // MARK: - Send events to Gist event listeners
+
+    // errorWithMessage
+    @MainActor
+    func test_givenAttemptToShowInlineMessage_EventListenerSet_expectFailToShow_expectRouteError() async {
+        messagingInAppImplementation.setEventListener(eventListenerMock)
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessageWithError(givenInlineMessage, insideOfInlineView: inlineView)
+
+        // Inline message does not display
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+
+        // errorWithMessage called
+        XCTAssertTrue(eventListenerMock.errorWithMessageCalled)
+        XCTAssertEqual(eventListenerMock.errorWithMessageCallsCount, 1)
+        XCTAssertEqual(eventListenerMock.errorWithMessageReceivedArguments, InAppMessage(gistMessage: givenInlineMessage))
+    }
+
+    @MainActor
+    func test_givenAttemptToShowInlineMessage_EventListenerNotSet_expectFailToShow_expectNoCallback() async {
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessageWithError(givenInlineMessage, insideOfInlineView: inlineView)
+
+        // Inline message does not display
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+
+        // errorWithMessage not called as eventListener is not set
+        XCTAssertFalse(eventListenerMock.errorWithMessageCalled)
+    }
+
+    // messageActionTaken when close button is tapped
```

Comment:
For modal messages today, [close buttons should not](https://github.com/customerio/customerio-ios/blob/2f4e4e4d07bedffc251b0f0a19b3e2bd55b0375d/Sources/MessagingInApp/MessagingInAppImplementation.swift#L97-L102) trigger a `messageActionTaken` call to the event listener. Inline messages have the same requirement and therefore, this test should fail.

Reply:
I expected the same but the code [does not seem to have any condition](https://github.com/customerio/customerio-ios/blob/2f4e4e4d07bedffc251b0f0a19b3e2bd55b0375d/Sources/MessagingInApp/MessagingInAppImplementation.swift#L97-L108) to prevent calling `messageActionTaken` event listener. It actually triggers from [here](https://github.com/customerio/customerio-ios/blob/1be401784e9467c1733ab06df9001c1f10f80eae/Sources/MessagingInApp/Gist/Managers/MessageManager.swift#L111), may be we would want to push a check [here](https://github.com/customerio/customerio-ios/blob/1be401784e9467c1733ab06df9001c1f10f80eae/Sources/MessagingInApp/Gist/Managers/MessageManager.swift#L111) to prevent handling button tap when close button is pressed to prevent it from happening. 🤔

Reply:
:facepalm: Sorry, my brain had a moment it looks like. 

As you said, there is no condition for preventing calling `messageActionTaken`. Ignore my comment.